src/storage/src/engine.rs

```diff
@@ -544,4 +553,99 @@ mod tests {
 
         assert!(engine.get_region(&ctx, "no such region").unwrap().is_none());
     }
+
+    #[tokio::test]
+    async fn test_drop_region() {
```

Comment:
nitpick: `test_create_new_region()` also needs to create a region.  Could we wrap a helper function to create the engine and the region from specific temp dirs?